TEST.md

```diff
@@ -194,6 +194,68 @@ To cleanup the namespaces after the test has finished, run:
 $ bin/test-cleanup
 ```
 
+### Testing the dashboard
+
+We use WebdriverIO to test how the web dashboard looks and operates locally in
+Chrome. For cross-browser testing, we use SauceLabs, which runs simulataneous
+tests on different browsers in the cloud.
+
+Whether you are running the tests locally or in the cloud, you'll first run
+`bin/web dev` which will start up the dashboard at `localhost:7777`.
+
+To run a local WebdriverIO instance that will run the tests on a local instance
+of Chrome, run:
+
+```bash
+$ bin/web integration local
+```
+
+To run cross-browser tests via SauceLabs, you need to do a few things first:
+
+1. Sign up for a (free) SauceLabs sub-account for the account 'buoyant'. If you
+   are not a Buoyant staffer, the best way to get an account invite is to ask in
+   the [Linkerd Slack channel](https://slack.linkerd.io).
+
+2. Once you have your username and key, set them as permanent environment
+   variables. This keeps your credentials private, and means that everyone on
+   the team can run the tests via their unique login without modifying the
+   authentication credentials in the test files. Open your `~/.bash_profile`
+   file and add:
+
+```
+export SAUCE_USERNAME="your Sauce username"
+export SAUCE_ACCESS_KEY="your Sauce access key"
+```
+
+3. Now you'll [download Sauce
+   Connect](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy), the
+   proxy server that will open a secure tunnel between a SauceLabs VM and the
+   Linkerd dashboard instance you're running on `localhost:7777`. You'll want to
+   save it in a separate directory than the rest of your development files. Once
+   you've downloaded it, navigate to its directory and run:
+
+```bash
+$ bin/sc -u $SAUCE_USERNAME -k $SAUCE_ACCESS_KEY
```

Comment:
The first time I saw this command, the leading `bin/` made me assume I was running it from this repo's dir. Consider a little more guidance, for example:

```bash
SC=sc-4.5.3-osx # for example on osx
wget -O - https://saucelabs.com/downloads/$SC.zip | tar xfz - -C ~/
cd ~/$SC
bin/sc -u $SAUCE_USERNAME -k $SAUCE_ACCESS_KEY
```

Related, these instructions are great (and totally work!). Down the road we could consider handling the download and execution of Sauce Connect, similar to how we do things in `bin/lint` or `bin/dep`.

Reply:
Hey @siggy, great feedback, thanks. I really like spelling out the process for the user so it’s as fast as possible. However I worry that by specifying a file location controlled by SauceLabs, they might change it at a later date and create confusion. What do we typically do in this type of situation?

Reply:
And THANKS for the `geckodriver` catch - that was inadvertently left in there from previous testing. It's definitely not supposed to be in there!

Reply:
@scottcarol Good question re: that `sc-4.5.3-osx`. I don't have a silver bullet answer. With both the `lint` and `dep` scripts (and `protoc`!), we also rely on a hard-coded version and URL. Looking at https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy, they do appear to host several older versions, so for now we just keep an eye on it. Once these tests get automated, we should know the moment that URL/version breaks.

In the ideal state, there would exist an API endpoint that would give us the latest version of the tool. `kubectl` does this:
```bash
curl -LO https://storage.googleapis.com/kubernetes-release/release/$(
  curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt
)/bin/darwin/amd64/kubectl
```
(from https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-binary-using-curl)

Our Linkerd CLI installer is similar, though slightly more magical:
```bash
curl https://run.linkerd.io/install-edge | sh
```

Reply:
@siggy Cool, thanks for this context.